### PR TITLE
[RLlib] RLlib deprecation Notices Part 1 (algorithm/, evaluation/, execution/, models/jax/)

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: Avnish <avnishnarayan@gmail.com>

This is part 1 of part 2 of deprecation notices that we'll put out for ray 2.6.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/apex_ddpg/apex_ddpg.py [hunk 1]
```diff
@@ -151,4 +152,16 @@ def setup(self, config: AlgorithmConfig):
     @override(DDPG)
     def training_step(self) -> ResultDict:
         """Use APEX-DQN's training iteration function."""
+        if log_once("apex-ddpg-deprecation-warning"):
+            deprecation_warning(
+                old="rllib/algorithms/apex_ddpg/",
+                new="rllib_contrib/apex_ddpg/",
+                help=(
+                    "This algorithm will be "
```
Comment (avnishn) [inline rllib/algorithms/apex_ddpg/apex_ddpg.py:160]: change to "This algorithm will be removed from rllib"
" It is being moved to the ray/rllib_contrib directory"

File: rllib/algorithms/mbmpo/mbmpo.py [hunk 2]
```diff
@@ -508,6 +509,18 @@ def get_default_policy_class(
     def execution_plan(
         workers: WorkerSet, config: AlgorithmConfig, **kwargs
     ) -> LocalIterator[dict]:
+        if log_once("mbmpo-deprecation-warning"):
+            deprecation_warning(
```
Comment (kouroshHakha) [inline rllib/algorithms/mbmpo/mbmpo.py:513]: error should be set to True?

File: rllib/execution/learner_thread.py [hunk 3]
```diff
@@ -65,6 +67,8 @@ def __init__(
         self.learner_info = {}
         self.stopped = False
         self.num_steps = 0
+        if log_once("learner-thread-deprecation-warning"):
+            deprecation_warning(old="ray.rllib.execution.learner_thread.LearnerThread")
```
Comment (kouroshHakha) [inline rllib/execution/learner_thread.py:71]: error = True?

**Review by kouroshHakha — COMMENTED** [review]
Should we raise error = True for anything that is related to execution plan?

**Review by kouroshHakha — APPROVED** [review]

**Pull request merged**